crates/router/src/core/payment_methods/cards.rs

```diff
@@ -316,18 +329,132 @@ pub async fn migrate_payment_method(
                 merchant_id.into(),
                 key_store,
                 merchant_account,
+                card_bin_details.clone(),
             )
             .await
         }
     }
 }
 
+pub async fn populate_bin_details_for_masked_card(
+    card_details: &api_models::payment_methods::MigrateCardDetail,
+    db: &dyn db::StorageInterface,
+) -> errors::CustomResult<api::CardDetailFromLocker, errors::ApiErrorResponse> {
+    helpers::validate_card_expiry(&card_details.card_exp_month, &card_details.card_exp_year)?;
+    let card_number = card_details.card_number.clone();
+
+    let (card_isin, last4_digits) = get_card_bin_and_last4_digits_for_masked_card(
+        card_number.peek(),
+    )
+    .change_context(errors::ApiErrorResponse::InvalidRequestData {
+        message: "Invalid card number".to_string(),
+    })?;
+
+    let card_bin_details = if card_details.card_issuer.is_some()
+        && card_details.card_network.is_some()
+        && card_details.card_type.is_some()
+        && card_details.card_issuing_country.is_some()
+    {
+        api::CardDetailFromLocker {
```

Comment:
write a from implementation for api::CardDetailFromLocker and pass required values to it. use CardDetailFromLocker::from instead of into which will help in going to the reference